### PR TITLE
Change `myOrders` query to accept multiple states

### PR DESCRIPTION
**Pull request by sepans**
#### Problem:

On the purchase history page, we don't want to show the `abandoned` orders but if we do the filtering on the client-side the pagination wouldn't work as expected so we need to do the filtering on the server-side,

#### Solution:

This will change the `state` params to `states` to be able to pass an array of states (everything except 'abandoned') from the client.

#### Breaking change 🚨 

This is a schema breaking change but since we haven't launched purchase history yet it is safe to synchronously deploy it

### CODE REVIEW

File: app/graphql/types/query_type.rb [hunk 1]
```diff
@@ -56,6 +56,8 @@ def orders(params = {})
   def my_orders(params = {})
     raise ActiveRecord::RecordNotFound unless context[:current_user][:id]
 
+    states = params.delete(:states)
+    params = params.merge(state: states) if states.present?
```
Comment (starsirius) [inline app/graphql/types/query_type.rb:60]: I think the new schema
```graphql
states: [OrderStateEnum!]
```
accepts `[]` as the states, which will get ignored here because it's not present. From the API perspective it makes sense to return nothing in this case, instead of all. Adding a test for the edge case helps here too!

Comment (sepans) [reply app/graphql/types/query_type.rb]: Great catch. Fixed in https://github.com/artsy/exchange/pull/641/commits/0924be25c3c2beba618f1f23b4fb3fb43d859c9b
